server/src/query/share.ts

```diff
@@ -100,41 +107,43 @@ export async function modifyContentSharedWith({
     }
   }
 
-  const contentIds = (
-    await prisma.$queryRaw<{ id: Uint8Array }[]>(Prisma.sql`
-    WITH RECURSIVE content_tree(id) AS (
-      SELECT id FROM content
-      WHERE id = ${contentId} AND ownerId = ${loggedInUserId} AND isDeletedOn IS NULL
-      UNION ALL
-      SELECT content.id FROM content
-      INNER JOIN content_tree AS ct
-      ON content.parentId = ct.id
-      WHERE content.isDeletedOn IS NULL
-    )
-
-    SELECT id from content_tree;
-    `)
-  ).map((obj) => obj.id);
-
-  if (contentIds.length === 0) {
-    throw new InvalidRequestError("Content does not exist or is not editable");
-  }
+  const descendantIds = await getDescendantIds(contentId, {
+    excludeAssignments: true,
+  });
+  const contentShareIds = [contentId, ...descendantIds];
 
-  const relevantContentShares = users.flatMap((userId) =>
-    contentIds.map((contentId) => ({ userId, contentId })),
+  const contentShares = users.flatMap((userId) =>
+    contentShareIds.map((contentShareId) => ({
+      userId,
+      // the specific content being shared, not the original, named `contentId`
+      contentId: contentShareId,
+      // `rootSharedId` is null if this is the original content being shared
+      rootSharedId: isEqualUUID(contentId, contentShareId) ? null : contentId,
```

Comment:
If the record already exists and `rootSharedId` is `null`, could we leave it at `null`? That way, content was already shared as a root, it doesn't disappear from the `Shared with me` list?